### PR TITLE
Fix constant-time token comparison leaking expected token length

### LINKED ISSUE #39: Constant-time token comparison still leaks token length
**Source:** Code Review CR-18

**File:** `src/server/http_server_transport.cpp:14-30`

When lengths differ, the dummy loop iterates `a.size()` times, not `max(a.size(), b.size())`. Timing still correlates with expected token length.

**Impact:** Minor timing side-channel on auth token length. Low severity in practice but inconsistent with the intent of constant-time comparison.

**Suggested fix:** Always iterate `max(a.size(), b.size())` times, or use a well-tested constant-time comparison implementation.

### PR DESCRIPTION
**Pull request by teaguesterling**
## Summary

Closes #39.

- **Fix `ConstantTimeEquals`** to use a fixed 256-byte buffer comparison instead of iterating `max(a.size(), b.size())` times. The previous approach still leaked the expected token length because an attacker could send tokens of increasing length and observe the timing transition point. The new implementation always iterates exactly 256 times regardless of input, eliminating the side-channel.
- **Add functional tests** (7b–7e) for token comparison edge cases: shorter token, longer token, same-length wrong token, and correct-prefix-wrong-suffix.
- **Add a Python reproducer** (`test/http/reproduce_timing_leak.py`) that demonstrates the timing leak by measuring response times across different attacker token lengths.

## Test plan

- [x] SQL logic tests pass (870 assertions, 23 test cases)
- [x] Integration tests pass (56/56)
- [x] HTTP transport tests pass (15/16; the 1 failure is a pre-existing version string mismatch unrelated to this PR)
- [ ] Optionally run `reproduce_timing_leak.py` against a server with the old vs new code to observe timing differences

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by teaguesterling — COMMENTED** [review]
## Security Review

The core approach — fixed 256-byte buffers with fixed iteration count — is a **solid improvement** over the previous `max(a.size(), b.size())` loop and correctly eliminates the length oracle described in #39.

### Compiler optimization robustness (non-blocking suggestion)

The `volatile unsigned char result` pattern is widely used and works on current GCC/Clang, but it only formally guarantees that reads/writes to `result` itself are not elided. The non-`volatile` source buffers (`buf_a`, `buf_b`) are theoretically subject to optimization under aggressive LTO/O3. In practice, the volatile accumulator creates a data dependency that prevents current compilers from eliminating the loop body, but this is a compiler implementation detail rather than a language guarantee.

Two options for hardening, neither blocking for this PR:
1. Make the buffers themselves `volatile` (requires byte-by-byte copy instead of `memcpy`)
2. Use a platform constant-time primitive like `CRYPTO_memcmp` (OpenSSL) or `timingsafe_memcmp` (glibc 2.25+/BSDs) — though this introduces portability considerations

### Other observations

- `.lq/commands.toml` — This is local build tooling config and probably shouldn't be included in this PR. Consider reverting it or splitting it out.
- No config-time validation that `auth_token` fits within the effective 249-byte limit (256 − 7 for `"Bearer "`). An unreasonably long token would silently fail all auth. Low priority but worth a bounds check when the token is configured.
- The Python reproducer is well-structured. Note that the 1.15x threshold may produce false results under network jitter — might be worth documenting that it works best on a quiet localhost.
- Tests 7b–7e are a good addition covering the important edge cases.

Overall: the algorithmic fix is correct and the PR is in good shape.

File: .lq/commands.toml [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-format = "make_error"
+description = "Build the extension (time consuming, don't use timeoutes)"
+format = "gcc_error,make_error,cmake_log"
 
 [commands.test]
 cmd = "make test"
-description = ""
-format = "make_error"
+description = "Run tests"
+format = "duckdb_tests"
```
[the remark this answers is not in the document]
Comment (teaguesterling) [reply .lq/commands.toml]: This file is identical between main and this branch — the diff appears to be an artifact of the GitHub comparison picking up a commit from the branch history that predates a merge into main. Running `git diff main..HEAD -- .lq/commands.toml` locally shows no difference. It should resolve cleanly on merge.
